pkg/apply/applier.go

```diff
@@ -368,6 +370,15 @@ type Options struct {
 	// that should be used for pruning. If this is not provided, the
 	// default is to use the Background policy.
 	PrunePropagationPolicy metav1.DeletionPropagation
+
+	// WaitForPrune defines whether we should wait for all resources
+	// to be fully deleted after pruning. This is only effective
+	// if the PrunePropagationPolicy is set to Forground.
+	WaitForPrune bool
+
+	// WaitForPruneTimeout defines how long we should wait for all
+	// resources to be deleted after prune before giving up.
+	WaitForPruneTimeout time.Duration
```

Comment:
Wondering if there is a way to test it.

Reply:
We do have tests that verify that these flags are used correctly in the solver to build the taskqueue. What we don't have is tests that verify that the flags are passed through. I can take a look and see if that is practical.
We do need to eventually spend some time to see if we can add better tests to the applier. It needs to have more of dependencies hidden behind interfaces so we can stub them out. I am planning to take a look at this after https://github.com/kubernetes-sigs/cli-utils/pull/153 is merged.

Reply:
I can't find a good way to test this with the current structure of the applier. Lets revisit this when we have better abstractions and interfaces to make unit testing easier.

Reply:
Sounds good. May be an e2e for same deployment with 1s timeout and 1m timeout and one succeeding and the other failing is achievable I guess. Might be an overkill but a thought.

Reply:
We want to support the same functionality in the destroy command, and then we should update all the e2e tests to verify that all resources have been deleted from the cluster after destroy.
Agree we should also verify that this works for prune in one of the e2e tests.